osquery/tables/system/windows/registry.cpp

```diff
@@ -263,8 +281,8 @@ void queryKey(const std::string& keyPath, QueryData& results) {
     if (regLinkStr != nullptr) {
       delete[](regLinkStr);
     }
+    delete[](bpDataBuff);
```

Comment:
Ah. Now I recall why I had the `ZeroMemory` above. What happens if any of the `if(!status.Ok())` checks above fail? I think this might never `delete[](bpDataBuff)`, and we'll leak memory. I believe I had this outside of the for loop and just resized as demanded to ensure we deallocated. Would you mind adding `delete[]`s before the `continue`s above?